tools/test/examples/examples_lib.py

```diff
@@ -494,4 +539,81 @@ def fetch_output_image(output):
             image = lines[index][7:]
             if os.path.isfile(image):
                 return image
-    return False
\ No newline at end of file
+    return False
+
+
+def _perform_post_build_checks(example_name, checks, elf_dir_path):
+    """
+    Perform post build checks.
+
+    The function raises a PostBuildCheckFailureError exceptions if an error is
+    encountered.
+    """
+    if not checks:
+        raise PostBuildCheckFailureError("No post build check specified")
+
+    # Find the elf file
+    elf_file = None
+    for dirpath, _, filenames in os.walk(elf_dir_path):
+        for filename in filenames:
+            if filename.endswith(".elf"):
+                elf_file = os.path.join(dirpath, filename)
+
+    if not elf_file:
+        raise PostBuildCheckFailureError(
+            "Cannot find ELF file in {}".format(elf_dir_path)
+        )
+
+    for check in set(checks):
+        SUPPORTED_POST_BUILD_CHECKS[check](example_name, elf_file)
+
+
+def _set_minimal_printf_floats_status(status):
+    """
+    Enable or disable floating point support in minimal-printf.
+
+    Pass the string `true` or `false` to enable or disable floating point.
+    """
+    with open(
+        os.path.join(ROOT, "platform", "mbed_lib.json"), "r"
+    ) as platform_lib_file:
+        data = json.load(platform_lib_file)
+
+    data["config"]["minimal-printf-enable-floating-point"]["value"] = status
+
+    with open(
+        os.path.join(ROOT, "platform", "mbed_lib.json"), "w"
+    ) as platform_lib_file:
+        json.dump(data, platform_lib_file, indent=4)
+
+
+# Post-build check functions should be listed below and must raise a
+# a PostBuildCheckFailureError in case of failure
+
+def _post_build_check_floating_point(example_name, elf_file):
+    """Check if there are floating points in the executable."""
+
+    if example_name in POST_BUILD_CHECK_FLOATS_EXCLUDED_EXAMPLES:
+        return
+
+    float_symbols = elf_float_checker.check_float_symbols(elf_file)
+
+    if float_symbols:
+        raise PostBuildCheckFailureError(
+            "Floating point symbols found in executable: {}".format(
+                float_symbols
+            )
+        )
+
+
+POST_BUILD_CHECK_FLOATS_EXCLUDED_EXAMPLES = [
```

Comment:
Whats the purpose of these? When should I add/remove example from here? Please leave a comment why are these being here (the definition for being excluded)

Reply:
https://github.com/ARMmbed/mbed-os/pull/11429/commits/c1f15a4cb0fa2eea9a9aa7a5ed9bd640cc5fa9de